### DISCUSSION TITLE
The env_file field in the docker-compose.yml causes the .devcontainer to fail on creation

### DISCUSSION THREAD

**Discussion by cysabi** [community/community#27158] [Codespaces]
<p>So here’s the most important part of the <code>devcontainer.json</code></p>
<pre><code class="lang-auto">	"dockerComposeFile": [
		"../docker-compose.yml",
		"docker-compose.yml"
	],
</code></pre>
<p>This came from using the <code>Existing Docker Compose (Extend)</code> option when creating remote container files.</p>
<p>This is part of my existing <code>docker-compose.yml</code></p>
<pre><code class="lang-auto">env_file: .env
</code></pre>
<p>However, as <code>.env</code> files are not committed to source, attempting to create a codespaces will actually cause a failure on build, resulting in an error about <code>no .env found</code></p>
<p>A possible solution would be to detect if this field is present somehow, and automatically have an empty temporary .env file</p>

**Comment by jkeech** [comment]
<p>There’s an <code>initializeCommand</code> that can run before the docker build/compose. In that command, you can seed an empty .env file if needed. You can also just remove the env file config if it’s always going to be empty. Codespaces will soon support user secrets that can you inject as environment variables into your codespaces, which will accomplish the same thing as the env file with providing the container access to secrets as environment variables that are not committed to source.</p>
<p>Hope that helps!</p>

**Comment by cysabi** [comment]
<p>Hi, thank you for your response! Where can I find the documentation about <code>initializeCommand</code>?</p>

**Comment by jkeech** [comment, marked as answer]
<p><a class="mention" href="/leptoflare">@LeptoFlare</a> you can find the docs at <a href="https://code.visualstudio.com/docs/remote/devcontainerjson-reference#_devcontainerjson-properties" class="inline-onebox">devcontainer.json reference</a>.</p>
<p>Here’s the description for the <code>initializeCommand</code></p>
<blockquote>
<p><code>initializeCommand</code> string, array<br>
A command string or list of command arguments to run on the <strong>local machine</strong> before the container is created. This runs either when the container image is being built and when the running container is created or started. The commands execute from the <code>workspaceFolder</code> locally. For example, <code>"yarn install"</code> . The array syntax <code>["yarn", "install"]</code> will invoke the command (in this case <code>yarn</code> ) directly without using a shell.<br>
:warning: The command is run wherever the source code is located. For Codespaces this is in the cloud.</p>
</blockquote>

**Comment by cysabi** [comment]
<p>I’ve set the initializeCommand to <code>["touch", ".env"]</code> Obviously this doesn’t work because it is at the root directory. But how can I make sure to create the file in the proper directory without resorting to hard-coding <code>/var/lib/docker/vsonlinemount/workspace/radia/.env</code><br>
<img src="https://user-images.githubusercontent.com/47790183/181102103-bc76265f-573d-4f78-906c-0b369e4d4e08.png" alt="image" data-base62-sha1="ga3WVa5g9IPZr7tGRNnRlsLIlF3" width="606" height="143"></p>

**Comment by jkeech** [comment]
<p>Looks like this is a bug. Sorry about that! The command is supposed to be executed from the repo root. We’ll work on a fix. In the mean time, you can hardcode it to the path you’ve referenced, but I agree that’s not a good long-term solution.</p>

**Comment by cysabi** [comment]
<p>Hardcoding it has worked, but the environment variables in the file are not being exported</p>
<p>I tried looking in the documentation of .devcontainer but I could not figure out how to get it to properly export</p>

**Comment by jkeech** [comment]
<p><a class="mention" href="/leptoflare">@LeptoFlare</a>, can you share the repo you are working on (or a different public repo that simulates the same issue)? What variables are you expecting to show up? Are you writing other values in the env file after touching it? If not, since it’s empty, I wouldn’t expect it to provide any additional values.</p>
<p>Sorry for all the questions, just trying to understand the scenario better and see what could be going on and how to fix it.</p>

**Comment by cysabi** [comment]
<p>Solution found, Basically you have to set up a debug profile, running the program using the integrated terminal will not source the .env file</p>
